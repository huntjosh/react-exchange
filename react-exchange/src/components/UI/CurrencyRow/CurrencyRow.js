import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CurrencyRow extends Component {
  render() {
    return (
      <div>
        Currency Row
      </div>
    );
  }
}

CurrencyRow.propTypes = {};

export default CurrencyRow;
