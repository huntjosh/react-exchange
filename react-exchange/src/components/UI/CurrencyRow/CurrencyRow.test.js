import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CurrencyRow from './CurrencyRow';

configure({ adapter: new Adapter() });
test('renders currency row', () => {
  const rate = {
    label: '2018-08-18',
    value: 1.2388,
  };

  const component = shallow(<CurrencyRow rate={rate} />);
  expect(component).toMatchSnapshot();
});
