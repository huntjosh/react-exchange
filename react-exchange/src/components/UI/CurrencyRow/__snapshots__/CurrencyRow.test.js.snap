// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders currency row 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyRow
    rate={
      Object {
        "label": "2018-08-18",
        "value": 1.2388,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Row
        gutter={0}
      >
        <Col
          offset={6}
          span={6}
        >
          2018-08-18
        </Col>
        <Col
          offset={6}
          span={6}
        >
          $
          1.2388
        </Col>
      </Row>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Col
            offset={6}
            span={6}
          >
            2018-08-18
          </Col>,
          <Col
            offset={6}
            span={6}
          >
            $
            1.2388
          </Col>,
        ],
        "gutter": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "2018-08-18",
            "offset": 6,
            "span": 6,
          },
          "ref": null,
          "rendered": "2018-08-18",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "$",
              1.2388,
            ],
            "offset": 6,
            "span": 6,
          },
          "ref": null,
          "rendered": Array [
            "$",
            1.2388,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Row
          gutter={0}
        >
          <Col
            offset={6}
            span={6}
          >
            2018-08-18
          </Col>
          <Col
            offset={6}
            span={6}
          >
            $
            1.2388
          </Col>
        </Row>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Col
              offset={6}
              span={6}
            >
              2018-08-18
            </Col>,
            <Col
              offset={6}
              span={6}
            >
              $
              1.2388
            </Col>,
          ],
          "gutter": 0,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "2018-08-18",
              "offset": 6,
              "span": 6,
            },
            "ref": null,
            "rendered": "2018-08-18",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "$",
                1.2388,
              ],
              "offset": 6,
              "span": 6,
            },
            "ref": null,
            "rendered": Array [
              "$",
              1.2388,
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
